packages/build-tools/src/utils/expoFingerprintCli.ts

```diff
@@ -0,0 +1,47 @@
+import resolveFrom, { silent as silentResolveFrom } from 'resolve-from';
+import spawnAsync from '@expo/turtle-spawn';
+import { BuildStepEnv } from '@expo/steps';
+
+export class ExpoFingerprintCLIModuleNotFoundError extends Error {}
+export class ExpoFingerprintCLIInvalidCommandError extends Error {}
+export class ExpoFingerprintCLICommandFailedError extends Error {}
+
+export async function expoFingerprintCommandAsync(
+  projectDir: string,
+  args: string[],
+  { env }: { env: BuildStepEnv }
+): Promise<string> {
+  let expoFingerprintCli;
+  try {
+    expoFingerprintCli =
+      silentResolveFrom(projectDir, '@expo/fingerprint/bin/cli') ??
+      resolveFrom(projectDir, '@expo/fingerprint/bin/cli.js');
```

Comment:
we should not directly resolve `@expo/fingerprint` from projectDir. that may break from pnpm isolated mode. we should resolve `projectDir -> expo -> `@expo/fingerprint`, or even re-export `expo/fingerprint-cli`

example for `projectDir -> expo -> `@expo/fingerprint` resolver
https://github.com/expo/expo/blob/ca42e1312eaf3238212ec437f7fa9583db78ea0c/packages/%40expo/fingerprint/src/ExpoVersions.ts#L20-L24

Reply:
Good call. I'll wait for a canary to be published with @quinlanj's change (making fingerprint a dependency of expo) since testing this is fairly complex otherwise.